src/components/Settings.tsx

```diff
@@ -1,18 +1,54 @@
 import * as React from "react";
-import {useContext} from "react";
+import {useContext, useState} from "react";
+import { Action, ActionTypes } from "../context/AppReducer";
 import {GlobalContext} from "../context/GlobalState";
 
 export function Settings() {
-    const {globalState } = useContext(GlobalContext);
+    const {globalState, dispatch } = useContext(GlobalContext);
+    const [tezosServer, setTezosServer] = useState(globalState.tezosServer);
+    const [apiKey, setApiKey] = useState(globalState.apiKey);
+    const [network, setNetwork] = useState(globalState.network);
+    const [address] = useState(globalState.address);
+    
+    const handleSettingsUpdateClick = () => {
+        const action: Action = {
+            type: ActionTypes.UpdateSettings,
+            newTezosServer: tezosServer,
+            newApiKey: apiKey,
+            newNetwork: network,
+            newAddress: address
+        }
+        dispatch(action);
+    }
+        
     return (
         <div id={"settings"}>
             <h1>Settings</h1>
             <p>Tezos Node</p>
-            <input id={"settings_tezosNode"} defaultValue={globalState.tezosServer} />
+
+            <input 
+            id={"settings_tezosNode"} 
+            defaultValue={globalState.tezosServer}
+            onChange={(e: React.FormEvent<HTMLInputElement>) => { setTezosServer(e.currentTarget.value)}}
+             />
+
             <p>Nautilus API Key</p>
-            <input id={"settings_apikey"} defaultValue={globalState.apiKey} />
+            <input 
+            id={"settings_apikey"} 
+            defaultValue={globalState.apiKey}
+            onChange={(e: React.FormEvent<HTMLInputElement>) => { setApiKey(e.currentTarget.value)}}
+            />
             <p>Network</p>
-            <input id={"settings_apikey"} defaultValue={globalState.network} />
+            <input 
+            id={"settings_apikey"} 
+            defaultValue={globalState.network}
+            onChange={(e: React.FormEvent<HTMLInputElement>) => { setNetwork(e.currentTarget.value)}}
+            />
+            <p>
```

Comment:
Please pull in the latest changes from `trunk` and fix all merge conflicts. We now also display derivation paths.